### PR TITLE
Batch retrieval integration tests

### PR DESCRIPTION
**Pull request by zhilingc**
Adding batch retrieval to our e2e test suite.

### CODE REVIEW

File: .prow/scripts/test-end-to-end.sh [hunk 1]
```diff
@@ -206,8 +278,17 @@ ORIGINAL_DIR=$(pwd)
 cd tests/e2e
 
 set +e
-pytest --junitxml=${LOGS_ARTIFACT_PATH}/python-sdk-test-report.xml
+pytest --allow_dirty true --junitxml=${LOGS_ARTIFACT_PATH}/python-sdk-test-report.xml
```
Comment (woop) [inline .prow/scripts/test-end-to-end.sh]: Why do we need this allow dirty to be true?

File: tests/e2e/test_e2e.py [hunk 2]
```diff
@@ -361,3 +385,15 @@ def test_large_volume(client):
             abs_tol=FLOAT_TOLERANCE,
         ):
             break
+
+    # Test if values in wh store are correct
+    feature_retrieval_job = batch_client.get_batch_features(
+        entity_rows=customer_data[['datetime', 'customer_id']],
+        feature_ids=[
+            "customer_transactions_large:1:daily_transactions",
+            "customer_transactions_large:1:total_transactions",
```
Comment (woop) [inline tests/e2e/test_e2e.py]: I really think we should be testing joins of multiple feature sets here. Can we please add that?

Comment (zhilingc) [reply tests/e2e/test_e2e.py]: See the test below this one

Comment (zhilingc) [reply tests/e2e/test_e2e.py]: That's a WIP, fixing some bugs wrt the joins before i push

Comment (woop) [reply tests/e2e/test_e2e.py]: Ok, should we wait for that or do you want to merge this in?

Comment (zhilingc) [reply tests/e2e/test_e2e.py]: No, please dont merge this in, I'm still fixing the batch retrieval

**Pull request closed without merging**